sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -530,6 +523,16 @@ class ResolveSessionCatalog(
         replace,
         viewType)
 
+    case ShowViewsStatement(SessionCatalogAndNamespace(_, ns), pattern) =>
+      val namespace = if (ns.isEmpty) {
+        None
+      } else if (ns.length == 1) {
+        Some(ns.head)
+      } else {
+        throw new AnalysisException(s"The database name is not valid: ${ns.quoted}")
```

Comment:
Plz add tests for the multi-part case.

Reply:
When creating database/namespace like `CREATE NAMESPACE ns1.ns2`, it throw invalid database name error. Does it related V2 Session Catalog or some other reason? Sorry for my lack of related knowledge.

Reply:
Yea, updated in 1d56c0e4fbe9a26232d6856a253371d16f24d4da, add negative test case.